Frameworks/CoreText/CTLine.mm

```diff
@@ -16,16 +16,38 @@
 
 #import <CoreText/CTLine.h>
 #import <StubReturn.h>
-#import "NSStringInternal.h"
 #import "CoreTextInternal.h"
 #import "CGContextInternal.h"
 #import "DWriteWrapper_CoreText.h"
 #import <CoreText/CTTypesetter.h>
 
 #import <algorithm>
 #import <numeric>
+#import <wchar.h>
 #import <vector>
 
+static CFStringRef _CTCreateReversedString(CFStringRef string) {
+    if (string == nullptr) {
+        return nullptr;
+    }
+
+    CFIndex length = CFStringGetLength(string);
+    CFIndex usedBufLen;
+    CFStringGetBytes(string, CFRangeMake(0, length), kCFStringEncodingUTF16, 0, false, nullptr, length, &usedBufLen);
```

Comment:
Yes it does take a faster way than if you provide a buffer.
Further improvements can be addressed by a bug to improve core text. The CR is mainly to remove Foundation dependency and what we have now is better than the earlier Foundation/copy based implementation.